src/plonk/circuit.rs

```diff
@@ -614,6 +687,13 @@ impl<F: Field> ConstraintSystem<F> {
         self.gates.push((name, poly));
     }
 
+    /// Allocate a new selector.
+    pub fn selector(&mut self) -> Selector {
```

Comment:
We might also want to add `Selector::new(&mut ConstraintSystem)` to match `Permutation::new` as in #199.